paper-server/src/main/java/org/bukkit/craftbukkit/entity/CraftCreaking.java

```diff
@@ -14,6 +20,34 @@ public Creaking getHandle() {
         return (Creaking) this.entity;
     }
 
+    @Override
+    public Location getHome() {
+        return CraftLocation.toBukkit(this.getHandle().getHomePos(), this.getHandle().level());
+    }
+
+    @Override
+    public void setHome(Location location) {
+        Preconditions.checkArgument(location != null, "location cannot be null");
+        Preconditions.checkArgument(this.getLocation().getWorld().equals(location.getWorld()), "location world need to be the same than the current world where is the creaking");
```

Comment:
`location world needs to be the same as the entity's current location`

Also, is this true? Does it blow up if different world?

Reply:
When make the PR in spigot just avoid test if try set a home in another world cause issues with behaviour... maybe can test if really cause issues...